VirtoCommerce.CartModule.Data/Repositories/CartRepositoryImpl.cs

```diff
@@ -183,28 +183,33 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
 
         public virtual ShoppingCartEntity[] GetShoppingCartsByIds(string[] ids, string responseGroup = null)
         {
-            var carts = ShoppingCarts.Where(x => ids.Contains(x.Id)).ToArray();
+            var result = new ShoppingCartEntity[0];
```

Comment:
It's not enough. If there are no carts you will still perform redundant requests.
Also there is no need to create an empty array if ids is not empty.
```csharp
ShoppingCartEntity[] result;

if (ids.IsNullOrEmpty())
{
    result = new ShoppingCartEntity[0];
}
else
{
    result = ShoppingCarts.Where(x => ids.Contains(x.Id)).ToArray();

    if (result.Any())
    {
        ids = result.Select(x => x.Id).ToArray();
        ...
    }
}

return result;
```